configtx/msp.go

```diff
@@ -74,6 +75,393 @@ type MSP struct {
+
+	err = msp.validateCACerts()
+	if err != nil {
+		return err
+	}
+
+	return msp.setConfig(m.configGroup)
+}
+
+// AddOUIdentifier adds a custom organizational unit identifier to the organization's MSP.
+func (m *OrganizationMSP) AddOUIdentifier(ou membership.OUIdentifier) error {
+	msp, err := getMSPConfig(m.configGroup)
+	if err != nil {
+		return err
+	}
+
+	for _, o := range msp.OrganizationalUnitIdentifiers {
+		if reflect.DeepEqual(o, ou) {
+			return nil
+		}
+	}
+
+	msp.OrganizationalUnitIdentifiers = append(msp.OrganizationalUnitIdentifiers, ou)
+
+	return msp.setConfig(m.configGroup)
+}
+
+// RemoveOUIdentifier removes an existing organizational unit identifier from the organization MSP.
+func (m *OrganizationMSP) RemoveOUIdentifier(ou membership.OUIdentifier) error {
+	msp, err := getMSPConfig(m.configGroup)
+	if err != nil {
+		return err
+	}
+
+	ous := msp.OrganizationalUnitIdentifiers[:]
+	for i, o := range msp.OrganizationalUnitIdentifiers {
+		if reflect.DeepEqual(o, ou) {
+			ous = append(ous[:i], ous[i+1:]...)
+			break
+		}
+	}
+
+	msp.OrganizationalUnitIdentifiers = ous
+
+	return msp.setConfig(m.configGroup)
+}
+
+// SetCryptoConfig sets the configuration for the cryptographic algorithms for the organization MSP.
+func (m *OrganizationMSP) SetCryptoConfig(cryptoConfig membership.CryptoConfig) error {
+	msp, err := getMSPConfig(m.configGroup)
+	if err != nil {
+		return err
+	}
+
+	msp.CryptoConfig = cryptoConfig
+
+	return msp.setConfig(m.configGroup)
+}
+
+// AddTLSRootCert adds a TLS root certificate trusted by the organization MSP.
+func (m *OrganizationMSP) AddTLSRootCert(cert *x509.Certificate) error {
+	msp, err := getMSPConfig(m.configGroup)
+	if err != nil {
+		return err
+	}
+
+	for _, c := range msp.TLSRootCerts {
+		if c.Equal(cert) {
+			return nil
+		}
+	}
+
+	msp.TLSRootCerts = append(msp.TLSRootCerts, cert)
+
+	err = msp.validateCACerts()
+	if err != nil {
+		return err
+	}
+
+	return msp.setConfig(m.configGroup)
+}
+
+// RemoveTLSRootCert removes a trusted TLS root certificate from the organization MSP.
+func (m *OrganizationMSP) RemoveTLSRootCert(cert *x509.Certificate) error {
+	msp, err := getMSPConfig(m.configGroup)
+	if err != nil {
+		return err
+	}
+
+	certs := msp.TLSRootCerts[:]
+	for i, c := range msp.TLSRootCerts {
+		if c.Equal(cert) {
+			certs = append(certs[:i], certs[i+1:]...)
+			break
+		}
+	}
+
+	msp.TLSRootCerts = certs
+
+	err = msp.validateCACerts()
+	if err != nil {
+		return err
+	}
+
+	return msp.setConfig(m.configGroup)
+}
+
+// AddTLSIntermediateCert adds a TLS intermediate cert trusted by the organization MSP.
+func (m *OrganizationMSP) AddTLSIntermediateCert(cert *x509.Certificate) error {
+	msp, err := getMSPConfig(m.configGroup)
+	if err != nil {
+		return err
+	}
+
+	for _, c := range msp.TLSIntermediateCerts {
+		if c.Equal(cert) {
+			return nil
+		}
+	}
+
+	msp.TLSIntermediateCerts = append(msp.TLSIntermediateCerts, cert)
+
+	err = msp.validateCACerts()
```

Comment:
Can you add a followup story for adding TLS intermediate cert validation up the chain to the `validateCACerts`?

Reply:
https://jira.hyperledger.org/browse/FAB-18099 added to the backlog.